### PR TITLE
Mkw hiv mort

### PR DESCRIPTION
**Pull request by mwalte10**
- Integrated the HIV natural history bit for children off treatment
- changed all instances of hC to hc per jeff's comment in leapfrog check in on 25/7
- made all paed model functions have 'child' as the second word to indicate that they are paediatric functions
- removed all instances of hard coding related to age
- separated hc1 from hc2 for output of the hiv stratified population

### CODE REVIEW

**Review by jeffeaton — CHANGES_REQUESTED** [review]
This looks good. Adding a couple of small comments.

Please merge #11 into this branch and then re-review.

File: inst/include/child_model_simulation.hpp [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        }
+
+      }
+    }
+  }
+
+  for (int g = 0; g < ss.num_genders; ++g) {
+    for (int hm = 0; hm < ss.hc1_disease_stages; ++hm) {
+      for (int hm_alt = 0; hm_alt < ss.hc2_disease_stages; ++hm_alt) {
+        for (int cat = 0 ; cat < ss.hTM; ++cat) {
```
[the remark this answers is not in the document]
Comment (mwalte10) [reply inst/include/child_model_simulation.hpp:38]: changing to hcTT for children Transmission Types

File: inst/include/child_model_simulation.hpp [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+                              IntermediateData<S, real_type> &intermediate) {
+  constexpr auto ss = StateSpace<S>();
+  const auto cpars = pars.children;
+
+  for (int g = 0; g < ss.num_genders; ++g) {
+    // Run only first 5 age groups in total population 0, 1, 2, 3, 4
+    for (int af = 0; af < ss.hc2_agestart; ++af) {
+      if (cpars.hc_nosocomial(time_step) > 0) {
+        // Divide by 10 because we want to evenly distribute over 2 genders and 5 age groups
+        state_next.infections(af, g) = cpars.hc_nosocomial(time_step) / (5.0 * ss.num_genders);
```
[the remark this answers is not in the document]
Comment (mwalte10) [reply inst/include/child_model_simulation.hpp]: updated

**Comment by mwalte10** [comment]
@r-ash here is the problem with the intermediate data

**Comment by r-ash** [comment]
Not sure about the code factor issues here, it is complaining about empty lines but is seemingly referencing an old commit. Those blank lines have been removed.

File: src/model_setup.h [hunk 3]
```diff
@@ -91,6 +91,26 @@ leapfrog::Parameters <real_type> setup_model_params(const Rcpp::List data,
...
+  //natural history parameters
+  const leapfrog::TensorMap3<double> hc1_cd4_mort = parse_data<double>(data, "paed_cd4_mort", ss.hc1_disease_stages, ss.hTM, ss.hc1_age_groups);
+  const leapfrog::TensorMap3<double> hc2_cd4_mort = parse_data<double>(data, "adol_cd4_mort", ss.hc2_disease_stages, ss.hTM, ss.hc2_age_groups);
+
+  const leapfrog::TensorMap1<double> hc1_cd4_prog = parse_data<double>(data, "paed_cd4_prog", ss.hc1_disease_stages);
+  const leapfrog::TensorMap1<double> hc2_cd4_prog = parse_data<double>(data, "adol_cd4_prog", ss.hc2_disease_stages);
+
+  //Not sure how to do just a single value, ASK ROB
+  // !!! TODO: Ask Rob about preferred approach to assigning a single value
+  const double ctx_effect = parse_data<double>(data, "ctx_effect", 1)(0);  // select the first element of a TensorMap1 and assign to double
```
[the remark this answers is not in the document]
Comment (r-ash) [reply src/model_setup.h]: Added support into the code generation to manage this.

File: inst/include/child_model_simulation.hpp [hunk 4]
```diff
@@ -7,45 +7,259 @@ namespace leapfrog {
...
+  static_assert(ModelVariant::run_child_model,
+                "run_hiv_child_infections can only be called for model variants where run_child_model is true");
+  constexpr auto ss = StateSpace<ModelVariant>().base;
+  constexpr auto hc_ss = StateSpace<ModelVariant>().children;
+
+  for (int s = 0; s < ss.NS; ++s) {
+    //less than 5 because there is a cd4 transition between ages 4 and 5
+    for (int a = 1; a < hc_ss.hc2_agestart; ++a) {
+      for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
+        for (int cat = 0; cat < hc_ss.hcTT; ++cat) {
```
Comment (r-ash) [inline inst/include/child_model_simulation.hpp:26]: Loop order here should be NS, hc1AG, hcTT, hc1DS

**Review by r-ash — APPROVED** [review]
This looks good to me, there are a few bits of commented out code which I would be tempted to remove if they're not needed anymore, or add a comment about why they are commented out.

**Pull request merged**